nixos/modules/services/misc/plikd.nix

```diff
@@ -0,0 +1,84 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.plikd;
+
+  format = pkgs.formats.toml {};
+
+  plikdCfg = format.generate "plikd.cfg" cfg.settings;
+in
+{
+  options = {
+    services.plikd = {
+      enable = mkEnableOption "Enable the plikd server";
+
+      user = mkOption {
+        type = types.str;
+        default = "plikd";
+        description = "User account under which Plikd runs.";
+      };
+
+      settings = mkOption {
+        type = format.type;
+        default = {
```

Comment:
You probably don't want to set a `default` here because if the sysadmin sets any value then all default values will be erased.

You probably want to set this in `config`, like so:
```
config = mkIf cfg.enable {
services.plikd.settings = {
  ...
};
}
```

Reply:
In fact I am not sure how to make it a default version, still allowing the admin to tweak it

Reply:
Sure take a look:
```
config = mkIf cfg.enable {
  services.plikd.settings = mapAttrs (name: mkDefault) {
    ListenPort = 8080;
    ListenAddress = "localhost";
    DataBackend = "file";
    ...
  };
}
```

Your values will be default but easily overridden by the sysadmin.

Reply:
Woa clever!

Reply:
I have a doubt, did you mean option or config ?

Reply:
I mean `config` as written above. You should move those default values into the `config` section.